src/components/Gauge.js

```diff
@@ -0,0 +1,263 @@
+      colors: [StyleConstants.Colors.PRIMARY].concat(d3.scale.category20().range()),
+      data: [],
+      dataPointColors: [StyleConstants.Colors.CHARCOAL].concat(d3.scale.category20b().range()),
+      dataPointRadius: 5,
+      dataPoints: [],
+      formatter (value) {
+        return value;
+      },
+      height: 150,
+      id: 'gauge',
+      opacity: 1,
+      padAngle: 0.02,
+      showBaseArc: true,
+      showDataLabel: true,
+      width: 150
+    };
+  },
+
+  getInitialState () {
+    return {};
+  },
+
+  componentWillMount () {
+    this._setupD3Functions(this.props);
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (!_isEqual(this.props.data, newProps.data)) {
+      this._setupD3Functions(newProps);
+    }
+  },
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return !_isEqual(this.props, nextProps) || !_isEqual(this.state, nextState);
+  },
+
+  _setupD3Functions (props) {
+    const dataSets = props.data.map(item => {
+      return item.value;
+    });
+    const startAngle = (-135 * (Math.PI / 180));
+    const endAngle = (135 * (Math.PI / 180));
+    const pie = d3.layout.pie().sort(null).padAngle(props.padAngle).endAngle(endAngle);
+    const values = pie(dataSets);
+    const radius = Math.min(props.width, props.height) / 2;
+    const standardArc = d3.svg.arc().outerRadius(radius - props.activeOffset).innerRadius(radius - props.arcWidth);
+    const hoveredArc = d3.svg.arc().outerRadius(radius).innerRadius(radius - props.arcWidth);
+    const baseArc = d3.svg.arc().outerRadius(radius - props.activeOffset).innerRadius(radius - props.arcWidth).startAngle(startAngle).endAngle(endAngle);
+
+    this.setState({
+      baseArc,
+      endAngle,
+      hoveredArc,
+      pie,
+      radius,
+      standardArc,
+      values
+    });
+  },
+
+  _renderArcs () {
+    const firstSegment = -135 * (Math.PI / 180);
+    const endOfFirstSegment = -90 * (Math.PI / 180);
+    const secondSegment = -90 * (Math.PI / 180);
+    const endOfSecondSegment = -45 * (Math.PI / 180);
+    const thirdSegment = -45 * (Math.PI / 180);
+    const endOfThirdSegment = 0;
+    const fourthSegment = 0;
+    const endOfFourthSegment = 45 * (Math.PI / 180);
+    const fifthSegment = 45 * (Math.PI / 180);
+    const endOfFifthSegment = 90 * (Math.PI / 180);
+    const sixthSegment = 90 * (Math.PI / 180);
+    const endOfSixthSegment = 135 * (Math.PI / 180);
+
+    const segments = [
+      { startAngle: firstSegment, endAngle: endOfFirstSegment, padAngle: 0.02 },
+      { startAngle: secondSegment, endAngle: endOfSecondSegment, padAngle: 0.02 },
+      { startAngle: thirdSegment, endAngle: endOfThirdSegment, padAngle: 0.02 },
+      { startAngle: fourthSegment, endAngle: endOfFourthSegment, padAngle: 0.02 },
+      { startAngle: fifthSegment, endAngle: endOfFifthSegment, padAngle: 0.02 },
+      { startAngle: sixthSegment, endAngle: endOfSixthSegment, padAngle: 0.02 }];
+
+    return segments.map((point, i) => {
```

Comment:
We don't use index for keys any more due to a tricky issue around React not knowing how to diff keyed elements correctly.  We now use a property from current object if possible and only fall back to using index if there is no better scenario.

In this case you can add a new key/value pair the the segments objects.  Something like `id` or `name` that you can then use for the key on the group element below.

```javascript
{ id: 'firstSegment', startAngle: firstSegment, endAngle: endOfFirstSegment, padAngle: 0.02 }

<g key={point.id}>
```